### ISSUE TITLE
At midnight smart remove deletes all snapshot taken the previous day

### ISSUE THREAD

**Issue by Xwang1976** [bit-team/backintime#1041]
I've set backintime to save a snapshot every 2 hours and then I've activated the smart backup to keep the snapshots taken the previous day, one snapshot a day for the last 14 days, one snapshot a week for the last 5 weeks and one snapshot a month for the last 2 months.
I expected that after a day it would have deleted only snapshots older than one day (if not needed for the daily, weekly and monthly rules) so that to have twelve snapshots taken in the last 24 hours.
Instead at midnight it has deleted all the snapshots taken the day before excep the first one.
Is it normal or is it a bug?
If it is normal how can I keep the last 12 snapshots (taken in the last 24 hours) and the daily, weekly and monthly ones?

**Comment by buhtzz** [comment]
Tag: Feedback, Support

@Xwang1976 There is a new forming maintaining team and we do review all issues. Is this problem still relevant for you or did you find a solution?

I need to summarize that do improve my understanding
- scheduled each 2 hours
- keep previous days "1"
- keep per day for the last "14" days
- keep one per week for the last "5" weeks
- keep one per month for the last "2" months

I am not sure if I understand everything. It seems normal to me what you subscribed.

Maybe you should set "keep previous days" to the value "2". Have you tried that?

**Comment by emtiu** [comment]
Yes, it sounds to me like the setting you want is: _"Keep all snapshots for the last: **2** day(s)"._

If I'm not mistaken, the confusion probably comes from the fact that backintime takes _"the last **1** day(s)"_ to mean _"the current calendar day"_ (which changes at midnight), and **not** _"the past 24 hours"_.

**Comment by buhtz** [comment]
Closing because of no new relevant information. Please feel free to re-open the issue.

**Comment by emtiu** [comment]
I'd like to keep this open and reproduce it some time, so we could maybe find a better wording for the smart remove settings.

**Comment by buhtz** [comment]
I used the "Documentation" label because IMHO the strings in the GUI are also kind of a documentation.

**Comment by Xwang1976** [comment]
> Yes, it sounds to me like the setting you want is: _"Keep all snapshots for the last: **2** day(s)"._
> 
> If I'm not mistaken, the confusion probably comes from the fact that backintime takes _"the last **1** day(s)"_ to mean _"the current calendar day"_ (which changes at midnight), and **not** _"the past 24 hours"_.

Sorry for the enormus delay, but indeed yes I would like to keep for last 2 days meaning that due to the fact that I do a bakup every 4 hours, I would like to keep the latest 12 snapshots the one for the last 48 hours. I hope it clarifies.

**Comment by buhtz** [comment]
Can you please review the [existing docu about remove & retention](https://backintime.readthedocs.io/en/latest/remove_retention.html) (aka Auto/Smart Remove) and give me some feedback.

Thanks in advance,
Christian

**Comment by buhtz** [comment]
It seems to me the issue is solved by improving the GUI and docu about smart remove. But I need some feedback here to be sure if this solves your problem.

Otherwise I need to close the issue.

**Comment by Xwang1976** [comment]
> It seems to me the issue is solved by improving the GUI and docu about smart remove. But I need some feedback here to be sure if this solves your problem.
> 
> Otherwise I need to close the issue.

The documentation is clear for me.

**Comment by buhtz** [comment]
Thank you for the feedback. I appreciate that.
And also thank you for contributing via reporting and discussing a problem.

**Comment by buhtz** [comment]
Closing this ticket based on the comment above. Feel free to reopen
if the problem still exists. Thank you for your efforts. If you have
any further questions, ideas or encounter any other issues, please
don't hesitate to let us know.

Best regards,
Christian
